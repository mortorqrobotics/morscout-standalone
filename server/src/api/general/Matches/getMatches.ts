import { Schema } from "mongoose";
import { Regional, Match } from "models";
export default async (regionalId: string) => {
  return await Match.find(
    {
      regional: regionalId
    },
    "+startDate"
  );
};
