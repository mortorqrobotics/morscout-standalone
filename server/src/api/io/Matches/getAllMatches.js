import { Schema } from "mongoose";
import { Team, Regional, User, Match } from "models";
export default async socket => {
  return [
    {
      teams: { blue: ["Hi"], red: ["Red"] },
      time: new Date(),
      progress: { max: 1, current: 1 }
    }
  ];
  // return await Match.find({
  //   _id: {
  //     $in: await Regional.findById(
  //       await Team.findById(await User.findById(socket.User.id).team).regional
  //     ).matches.map(id => Schema.Types.ObjectId(id))
  //   }
  // });
};
