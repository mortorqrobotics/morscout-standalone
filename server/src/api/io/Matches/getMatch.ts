import { Match } from "models";
import { Schema } from "mongoose";
export default async (id: Schema.Types.ObjectId) => {
  return await Match.findById(id);
};
