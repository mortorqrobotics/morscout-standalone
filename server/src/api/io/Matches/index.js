import { updated, loadAllMatches } from "Shared/types/Matches";
import getMatches from "./getAllMatches";

export default socket => {
  socket.on(loadAllMatches, async () => {
    socket.emit("action", {
      type: updated,
      data: await getMatches(socket)
    });
  });
};
