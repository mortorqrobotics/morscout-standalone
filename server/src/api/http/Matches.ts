import getMatches from "api/Matches/getMatches";
import getMatch from "api/Matches/getMatch";
import { Router, Request, Response } from "express";

export default (router: Router) => {
  router.get("/matches/:regional", async (req: Request, res: Response) => {
    res.end(JSON.stringify(await getMatches(req.params.regional)));
  });
  router.get("/match/:match", async (req: Request, res: Response) => {
    res.end(JSON.stringify(await getMatch(req.params.match)));
  });
};
