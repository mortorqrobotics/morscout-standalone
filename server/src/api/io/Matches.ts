import { updated, loadAllMatches, loadMatch } from "shared/types/matches";
import getMatches from "api/matches/getMatches";
import getMatch from "api/matches/getMatch";
import { Schema } from "mongoose";
import { Socket } from "socket.io";

export default (socket: Socket) => {
  socket.on(loadAllMatches, async (regionalId: string) => {
    socket.emit("action", {
      type: updated,
      data: await getMatches(regionalId)
    });
  });
  socket.on(loadMatch, async (id: string) => {
    socket.emit("action", {
      type: updated,
      data: {
        [id.toString()]: await getMatch(id)
      }
    });
  });
};
