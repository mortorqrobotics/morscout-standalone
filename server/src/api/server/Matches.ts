import getMatches from "api/Matches/getMatches";
import getMatch from "api/Matches/getMatch";
import { Router, Request, Response } from "express";

export default (router: Router) => {
  router.get("/matches/:season", async (req: Request, res: Response) => {
    res.send(JSON.stringify(await getMatches(req.params.season)));
  });
  router.get("/match/:team", async (req: Request, res: Response) => {
    res.send(JSON.stringify(await getMatch(req.params.team)));
  });
};
