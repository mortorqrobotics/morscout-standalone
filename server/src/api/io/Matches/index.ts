import { updated, loadAllMatches, loadMatch } from "Shared/types/Matches";
import getMatches from "./getMatches";
import getMatch from "./getMatch";
import { Schema } from "mongoose";
import { Socket } from "socket.io";

export default (socket: Socket) => {
  socket.on(loadAllMatches, async () => {
    socket.emit("action", {
      type: updated,
      data: await getMatches(socket)
    });
  });
  socket.on(loadMatch, async (id: Schema.Types.ObjectId) => {
    socket.emit("action", {
      type: updated,
      data: {
        [id.toString()]: await getMatch(id)
      }
    });
  });
};
