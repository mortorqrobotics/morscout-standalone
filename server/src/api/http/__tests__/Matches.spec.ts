import express from "express";
import request from "supertest";
import { setup, teardown } from "setup/mongoose";

import { Regional, Match } from "models";

import Matches from "../Matches";

let router: express.Router;

beforeAll(setup);
afterAll(teardown);

beforeEach(async () => {
  router = express.Router();
  jest.spyOn(router, "get");
  Matches(router);
});

test("matches has the same routes", async () => {
  expect(
    (router.get as jest.Mock).mock.calls.map(val => val[0]).sort()
  ).toMatchSnapshot();
});

test("get matches by regional works", async () => {
  const regional = await Regional.findOne();
  const response = await request(router).get(`/matches/${regional._id}`);
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
});

test("get matches by id works", async () => {
  const regional = await Match.findOne();
  const response = await request(router).get(`/match/${regional._id}`);
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
});
