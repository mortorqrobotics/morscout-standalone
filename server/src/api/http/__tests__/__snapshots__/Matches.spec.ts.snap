// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get matches by id works 1`] = `"{\\"_id\\":\\"7d7be5a7994e35048b9e5e0e\\",\\"redAlliance\\":\\"2d7ce5a7994e35048b9e5e0e\\",\\"blueAlliance\\":\\"2d7ce5a7994e35048b9e5e0e\\",\\"winner\\":\\"Red\\",\\"startDate\\":\\"2019-02-03T17:00:00.000Z\\",\\"endDate\\":\\"2019-02-03T17:10:00.000Z\\",\\"Regional\\":\\"6d7be5a7994e35048b9e5e0e\\",\\"__v\\":0}"`;

exports[`get matches by regional works 1`] = `"[{\\"_id\\":\\"7d7be5a7994e35048b9e5e0e\\",\\"redAlliance\\":\\"2d7ce5a7994e35048b9e5e0e\\",\\"blueAlliance\\":\\"2d7ce5a7994e35048b9e5e0e\\",\\"winner\\":\\"Red\\",\\"startDate\\":\\"2019-02-03T17:00:00.000Z\\",\\"endDate\\":\\"2019-02-03T17:10:00.000Z\\",\\"Regional\\":\\"6d7be5a7994e35048b9e5e0e\\",\\"__v\\":0}]"`;

exports[`matches has the same routes 1`] = `
Array [
  "/match/:match",
  "/matches/:regional",
]
`;
